src/transaction.cpp

```diff
@@ -903,20 +920,11 @@ bool wsrep::transaction::bf_abort(
         {
             // We need to change the state to aborting under the
             // lock protection to avoid a race between client thread,
-            // otherwise it could happend that the client gains control
+            // otherwise it could happen that the client gains control
             // between releasing the lock and before background
             // rollbacker gets control.
             state(lock, wsrep::transaction::s_aborting);
             client_state_.set_rollbacker_active(true);
-
-            if (client_state_.mode() == wsrep::client_state::m_high_priority)
-            {
-                lock.unlock();
-                client_state_.server_state().stop_streaming_applier(
```

Comment:
Didn't it cause a memory leak? It removes the streaming applier pointer for the map, so no possibility to delete it afterward.